nmt.go

```diff
@@ -11,6 +11,12 @@ import (
 	"github.com/celestiaorg/nmt/namespace"
 )
 
+const (
+	DefaultNamespaceIDLen = 8
+	DefaultShareSize      = 512
```

Comment:
Might be worth calling it LeafSiae as Share is a concept outside of the NMT and there is a notion of Leaf already in NMT.